modules/currency-l0/src/main/scala/io/constellationnetwork/currency/l0/snapshot/CurrencySnapshotConsensusStateAdvancer.scala

```diff
@@ -277,7 +277,9 @@ object CurrencySnapshotConsensusStateAdvancer {
                     ) =>
                   {
                     for {
-                      maybeAllBinarySignatures <- OptionT.liftF(maybeGetAllDeclarations(state, resources, config)(_.binarySignature))
+                      maybeAllBinarySignatures <- OptionT.liftF(
+                        maybeGetAllDeclarations(state, resources, config)(_.binarySignature, _.signaturesLatestUnique)
+                      )
                       binarySignatures <- OptionT.fromOption[F](maybeAllBinarySignatures)
```

Comment:
^ Also need to double check this part; not sure I accounted for this correctly in the addSignature portion ?